cmd/trtl/main.go

```diff
@@ -530,6 +583,607 @@ func initPeersClient(c *cli.Context) (err error) {
+		}
+		if status, ok := grpc.FromError(err); !ok || status.Code() != codes.NotFound {
+			results = multierror.Append(results, err)
+			continue recordLoop
+		}
+
+		// Parse the issued at timestamp
+		var issuedAt time.Time
+		const layout = "2006-01-02T15:04:05.000000"
+		if issuedAt, err = time.Parse(layout, record[rows["Issued At"]]); err != nil {
+			results = multierror.Append(results, fmt.Errorf("record %d has invalid issued at time: %s", i, err))
+			continue recordLoop
+		}
+		expiresAt := issuedAt.AddDate(0, 13, 0)
+		if expiresAt.After(time.Now()) {
+			cert.Status = models.CertificateState_EXPIRED
+		}
+
+		// Parse the subject params into the subject and issuer
+		subject := &gds.Name{}
+		issuer := &gds.Name{
+			CommonName: "CipherTrace Issuing CA",
+		}
+		var profileId string
+		if profileId = record[rows["Subject"]]; profileId == "" {
+			results = multierror.Append(results, fmt.Errorf("record %d has no subject params", i))
+			continue recordLoop
+		}
+		params := strings.Split(profileId, ",")
+		if len(params) == 1 {
+			subject.CommonName = params[0]
+		} else {
+			// Parse the individual profile params
+			for _, param := range params {
+				kv := strings.Split(param, "=")
+				if len(kv) != 2 {
+					results = multierror.Append(results, fmt.Errorf("record %d has invalid profile param: %s", i, param))
+					continue recordLoop
+				}
+				switch kv[0] {
+				case "CN":
+					subject.CommonName = kv[1]
+				case "O":
+					subject.Organization = []string{kv[1]}
+					issuer.Organization = []string{kv[1]}
+				case "L":
+					subject.Locality = []string{kv[1]}
+					issuer.Locality = []string{kv[1]}
+				case "ST":
+					subject.Province = []string{kv[1]}
+					issuer.Province = []string{kv[1]}
+				case "C":
+					subject.Country = []string{kv[1]}
+					issuer.Country = []string{kv[1]}
+				default:
+					results = multierror.Append(results, fmt.Errorf("record %d has unknown profile param: %s", i, param))
+					continue recordLoop
+				}
+			}
+		}
+
+		if subject.CommonName == "" {
+			results = multierror.Append(results, fmt.Errorf("could not parse common name from record %d", i))
+			continue recordLoop
+		}
+
+		// Get the VASP from the common name
+		var vasp *gds.VASP
+		var ok bool
+		if vasp, ok = vasps[subject.CommonName]; !ok {
+			results = multierror.Append(results, fmt.Errorf("could not find VASP with common name %s", subject.CommonName))
+			continue recordLoop
+		}
+
+		// Get the certificate request from the batch ID
+		var batchId int64
+		if batchId, err = strconv.ParseInt(record[rows["Batch Id"]], 10, 64); err != nil {
+			results = multierror.Append(results, fmt.Errorf("record %d has invalid batch id: %s", i, err))
+			continue recordLoop
+		}
+		var certReq *models.CertificateRequest
+		if certReq, ok = certReqs[batchId]; !ok {
+			cert.Status = models.CertificateState_REVOKED
+			results = multierror.Append(results, fmt.Errorf("WARNING: could not find certificate request with batch id %d", batchId))
```

Comment:
Not sure what's going on here, I was hoping to use the batch ID to lookup the corresponding certificate request but it seems that for some of the certificates (a few alice and bob ones, as well as a traveler staging cert) to be restored, there is no matching certificate request for the batch ID. Have they been deleted? If so should we assume they are revoked?